pkg/app/piped/cloudprovider/kubernetes/state.go

```diff
@@ -351,9 +351,11 @@ func determineJobHealth(obj *unstructured.Unstructured) (status model.Kubernetes
 			failed = true
 			completed = true
 			message = condition.Message
+			break
 		case batchv1.JobComplete:
 			completed = true
 			message = condition.Message
+			break
```

Comment:
It shouldn't break here because what we want to do here is check if any one of the conditions gets failed, 

Here is a comment of `JobStatus.Conditions`.

>	// The latest available observations of an object's current state. When a Job
	// fails, one of the conditions will have type "Failed" and status true. When

Reply:
Ref: https://github.com/kubernetes/api/blob/master/batch/v1/types.go#L202-L203